File: src/main/resources/static/app/publication-work-details/ratings-service.js
$(function () {
    function ratings(data) {
        function ratePublicationWork(publicationWorkId, value) {
            return data.post('rate', { publicationWorkId: publicationWorkId, value: value });
        }

        return {
            ratePublicationWork: ratePublicationWork
        };
    }

    angular.module('librarySystem.services')
        .factory('ratings', ['data', ratings]);
}());

File: src/main/java/com/librarysystem/services/RatingsService.java
package com.librarysystem.services;

import com.librarysystem.data.models.Rating;

public interface RatingsService {
    Double getAverageRatingOfPublicationWork(long publicationWorkId);

    void save(Rating rating);

    Byte getRatingForPublicationWorkByIp(long publicationWorkId, String ip);
}
